File: DevSupportAndroid/src/main/java/com/sup/dev/android/views/widgets/ViewErrorEditText.java
package com.sup.dev.android.views.widgets;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.util.AttributeSet;

public class ViewErrorEditText extends android.support.design.widget.TextInputEditText {

    public ViewErrorEditText(Context context) {
        super(context);
    }

    public ViewErrorEditText(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public ViewErrorEditText(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    public void setError(boolean b){
        setError(b?"":null);
    }

    @Override
    public void setError(CharSequence pError, Drawable pIcon) {
        setCompoundDrawables(null, null, pIcon, null);
    }

}

File: DevSupportAndroid/src/main/java/com/sup/dev/android/views/views/ViewErrorEditText.kt
package com.sup.dev.android.views.views

import android.content.Context
import android.graphics.drawable.Drawable
import android.util.AttributeSet


open class ViewErrorEditText : android.support.v7.widget.AppCompatEditText {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle)

    override fun setError(pError: CharSequence?, pIcon: Drawable?) {
        when {
            pError == null -> super.setError(null, null)
            pError.isEmpty() -> setCompoundDrawables(null, null, pIcon, null)
            else -> super.setError(pError, pIcon)
        }
    }

}
